mlflow/sagemaker/__init__.py

```diff
@@ -157,8 +170,35 @@ def deploy(app_name, model_path, execution_role_arn=None, bucket=None, run_id=No
     :param run_id: MLflow run id.
     :param image: name of the Docker image to be used. if not specified, uses a 
                   publicly-available pre-built image.
-    :param region_name: Name of the AWS region to deploy to.
+    :param region_name: Name of the AWS region to which to deploy the application.
+    :param mode: The mode in which to deploy the application. Must be one of the following:
+
+                 mlflow.sagemaker.DEPLOYMENT_MODE_CREATE: Creates an application with the specified 
+                     name and model. This will fail if an application of the same name already 
+                     exists.
+
+                 mlflow.sagemaker.DEPLOYMENT_MODE_REPLACE: Creates an application with the 
+                     specified name and model. This will replace any pre-existing applications of
+                     the same name.
+
+                 mlflow.sagemaker.DEPLOYMENT_MODE_ADD: Adds the specified model to a pre-existing 
+                     application with the specified name, if one exists. If the application does 
+                     not exist,  a new application will be created with the specified name and 
+                     model. NOTE: The specified model will be added to the application's 
+                     corresponding SageMaker endpoint with an initial weight of zero (0). To route
+                     traffic to the model, update the application's associated endpoint
+                     configuration using either the AWS console or the
+                     use the `UpdateEndpointWeightsAndCapacities` function defined in the SageMaker 
+                     API Documentation
+                     (https://docs.aws.amazon.com/sagemaker/latest/dg/API_UpdateEndpointWeightsAndCapacities.html).
+
+    :param archive: If True, any pre-existing SageMaker application resources that become inactive
+                    (i.e. as a result of deploying in mlflow.sagemaker.DEPLOYMENT_MODE_REPLACE mode) 
+                    will be preserved. If False, these resources will be deleted.
     """
+    if mode not in DEPLOYMENT_MODES:
+        raise Exception("`mode` must be one of: {mds}".format(mds=",".join(DEPLOYMENT_MODES)))
```

Comment:
`ValueError` is the conventional Exception to use for when an argument value is not valid.